File: baekjoon/1764.py
# 3 4
# ohhenrie
# charlie
# baesangwook
# obama
# baesangwook
# ohhenrie
# clinton

import sys
input = sys.stdin.readline
n, m = map(int, input().split())

a = set()
b = set()

for i in range(n):
    a.add(input().strip())

for i in range(m):
    b.add(input().strip())

result = sorted(a & b)
print(len(a & b))
for i in result:
    print(i)

File: javascript/1764.js
// 3 4
// ohhenrie
// charlie
// baesangwook
// obama
// baesangwook
// ohhenrie
// clinton

const fs = require("fs");
const io = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = fs.readFileSync(io).toString().trim().split("\n");

const [N, M] = input.shift().split(" ").map(Number);

/**
 * Set
 * 중복되지 않는 값들의 집합입니다.
 * 각 값은 한 번만 등장할 수 있습니다.
 * 키-값 쌍이 아닌 값만 저장합니다.
 * 주요 메서드: add(), has(), delete(), clear()
 */

console.log(c);
const a = new Set(input.slice(0, N));
const b = new Set(input.slice(N));

const result = [];
for (let name of b) {
  if (a.has(name)) result.push(name);
}

console.log(result.length);
console.log(result.sort().join("\n"));
